parsl/executors/high_throughput/mpi_prefix_composer.py

```diff
@@ -0,0 +1,135 @@
+import logging
+from typing import Dict, List, Tuple, Set
+
+logger = logging.getLogger(__name__)
+
+VALID_LAUNCHERS = ('srun',
+                   'aprun',
+                   'mpiexec')
+
+
+class InvalidResourceSpecification(Exception):
+    """Exception raised when Invalid keys are supplied via resource specification"""
+
+    def __init__(self, invalid_keys: Set[str]):
+        self.invalid_keys = invalid_keys
+
+    def __repr__(self):
+        return f"Invalid resource specification options supplied: {self.invalid_keys}"
+
+    def __str__(self):
+        return self.__repr__()
+
+
+def validate_resource_spec(resource_spec: Dict[str, str]) -> bool:
```

Comment:
this should return `None` with an invalid state being signalled by an exception. *or* it should return bool and not raise exceptions

Reply:
Fair enough, I've updated this method.